### PR TITLE
Add NetworkInterface

### LINKED ISSUE #120: Implement NetworkInterface
As oneview-redfish-user, I access "/redfish/v1/Systems/<server_hardware_uuid>/NetworkInterfaces/\<deviceNumber\>" and receive a NetworkInterface object associated with this device at OneView's Server Hardware. There should be a link to a corresponding NetworkAdapter using the same deviceNumber as ID. It should also point to collections of NetworkPort and NetworkDeviceFunction objects.

### PR DESCRIPTION
**Pull request by ffdarkpenguin**
closes #120

### CODE REVIEW

File: oneview_redfish_toolkit/api/network_interface.py [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+            from Oneview
+
+            Args:
+                server_hardware: Oneview's server hardware dict
+        """
+        super().__init__(self.SCHEMA_NAME)
+        self.redfish["@odata.type"] = \
+            "#NetworkInterface.v1_1_0.NetworkInterface"
+        self.redfish["Id"] = device_id
+        index = int(device_id) - 1
```
Comment (llvieira) [inline oneview_redfish_toolkit/api/network_interface.py]: What if device_id is an invalid value? An **IndexError** will be raised. I'm considering this case in NetworkPortCollection.

File: oneview_redfish_toolkit/blueprints/network_interface.py [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+from hpOneView.exceptions import HPOneViewException
+from oneview_redfish_toolkit.api.network_interface import NetworkInterface
+from oneview_redfish_toolkit import util
+
+
+network_interface = Blueprint("network_interface", __name__)
+
+
+@network_interface.route(
+    "/redfish/v1/Systems/<uuid>/NetworkInterfaces/<device_id>",
```
Comment (llvieira) [inline oneview_redfish_toolkit/blueprints/network_interface.py]: Missing **/** at the end of the route.

Comment (llvieira) [reply oneview_redfish_toolkit/blueprints/network_interface.py]: You can change .../<device_id> to .../<int:device_id>, so the cast is no longer needed.

File: oneview_redfish_toolkit/blueprints/network_interface_collection.py [hunk 3]
```diff
@@ -32,7 +32,7 @@
 
 
 @network_interface_collection.route(
-    "/redfish/v1/Systems/<uuid>/NetworkInterfaces", methods=["GET"])
+    "/redfish/v1/Systems/<uuid>/NetworkInterfaces/", methods=["GET"])
 def get_network_interface_collection(uuid):
```
Comment (llvieira) [inline oneview_redfish_toolkit/blueprints/network_interface_collection.py]: Missing {} in logging in line 70.

File: oneview_redfish_toolkit/tests/api/test_network_interface.py [hunk 4]
```diff
@@ -0,0 +1,80 @@
...
+            self.server_hardware = json.load(f)
+
+        # Loading NetworkInterface mockup result
+        with open(
+            'oneview_redfish_toolkit/mockups_redfish/'
+            'NetworkInterface3.json'
+        ) as f:
+            self.network_interface_mockup = f.read()
+
+        self.device_id = "3"
```
Comment (llvieira) [inline oneview_redfish_toolkit/tests/api/test_network_interface.py]: If you change the route you can pass an integer here.

File: oneview_redfish_toolkit/api/network_interface.py (hunk 1 above)
Comment (llvieira) [reply oneview_redfish_toolkit/api/network_interface.py]: If you change the route this cast is no longer needed.

File: oneview_redfish_toolkit/blueprints/network_interface.py [hunk 5]
```diff
@@ -0,0 +1,79 @@
...
+
+        Return NetworkInterface Redfish JSON for a given hardware UUID
+        and device_id.
+
+        Parameters:
+            uuid: the UUID of the server_hardware
+            device_id: The id of the network device
+
+        Returns:
+            JSON: Redfish json with Thermal
```
Comment (danielreed) [inline oneview_redfish_toolkit/blueprints/network_interface.py]: Update this comment

Comment (ffdarkpenguin) [reply oneview_redfish_toolkit/blueprints/network_interface.py]: fixed

File: oneview_redfish_toolkit/blueprints/network_interface.py (hunk 2 above)
Comment (ffdarkpenguin) [reply oneview_redfish_toolkit/blueprints/network_interface.py]: If I do that I need to cast it back to string in other places. Going to keep it a string.
Will do validation here not in the class.

File: oneview_redfish_toolkit/blueprints/network_interface_collection.py (hunk 3 above)
Comment (ffdarkpenguin) [reply oneview_redfish_toolkit/blueprints/network_interface_collection.py]: fixed

File: oneview_redfish_toolkit/tests/api/test_network_interface.py (hunk 4 above)
Comment (ffdarkpenguin) [reply oneview_redfish_toolkit/tests/api/test_network_interface.py]: will not change...

File: oneview_redfish_toolkit/api/network_interface.py [hunk 6]
```diff
@@ -0,0 +1,66 @@
...
+
+        self.redfish["NetworkDeviceFunctions"] = dict()
+        self.redfish["NetworkDeviceFunctions"]["@odata.id"] = \
+            "/redfish/v1/Chassis/" + server_hardware["uuid"] + \
+            "/NetworkAdapter/" + device_id + "/NetworkDeviceFunctions/"
+
+        self.redfish["@odata.context"] = \
+            "/redfish/v1/$metadata#NetworkInterface.NetworkInterface"
+        self.redfish["@odata.id"] = "/redfish/v1/Systems/" + \
+            server_hardware["uuid"] + "/NetworkInterface/" + device_id
```
Comment (ricardoas) [inline oneview_redfish_toolkit/api/network_interface.py]: NetworkInterface**s**

File: oneview_redfish_toolkit/mockups_redfish/NetworkInterface3.json [hunk 7]
```diff
@@ -0,0 +1,18 @@
...
+        }
+    },
+    "NetworkPorts": {
+        "@odata.id": "/redfish/v1/Chassis/30303437-3034-4D32-3230-313133364752/NetworkAdapter/3/NetworkPorts/"
+    },
+    "NetworkDeviceFunctions": {
+        "@odata.id": "/redfish/v1/Chassis/30303437-3034-4D32-3230-313133364752/NetworkAdapter/3/NetworkDeviceFunctions/"
+    },
+    "@odata.context": "/redfish/v1/$metadata#NetworkInterface.NetworkInterface",
+    "@odata.id": "/redfish/v1/Systems/30303437-3034-4D32-3230-313133364752/NetworkInterface/3"
```
Comment (ricardoas) [inline oneview_redfish_toolkit/mockups_redfish/NetworkInterface3.json]: NetworkInterface**s**

File: oneview_redfish_toolkit/api/network_interface.py (hunk 6 above)
Comment (ffdarkpenguin) [reply oneview_redfish_toolkit/api/network_interface.py]: fixed

File: oneview_redfish_toolkit/mockups_redfish/NetworkInterface3.json (hunk 7 above)
Comment (ffdarkpenguin) [reply oneview_redfish_toolkit/mockups_redfish/NetworkInterface3.json]: fixed

File: oneview_redfish_toolkit/api/network_interface.py (hunk 1 above)
Comment (ffdarkpenguin) [reply oneview_redfish_toolkit/api/network_interface.py]: Adding validation in the blueprint

**Pull request merged**